assets/js/base/components/cart-checkout/place-order-button/index.js

```diff
@@ -2,23 +2,33 @@
  * External dependencies
  */
 import { useCheckoutContext } from '@woocommerce/base-context';
+import PropTypes from 'prop-types';
 
 /**
  * Internal dependencies
  */
 import Button from '../button';
 
-const PlaceOrderButton = () => {
+const PlaceOrderButton = ( { validateSubmit } ) => {
 	const { submitLabel, onSubmit } = useCheckoutContext();
 
 	return (
 		<Button
 			className="wc-block-components-checkout-place-order-button"
-			onClick={ onSubmit }
+			onClick={ () => {
+				const isValid = validateSubmit();
+				if ( isValid ) {
```

Comment:
The idea I had was that checkout context would register a default callback with the `onCheckoutProcessing` event subscriber and then the callback would return the error object with the validation errors in it. Then checkout can take care of setting validation errors based on that response and abort the checkout setting checkout status appropriately etc. 

That way our components only have to:

- output a validation error if there is one in the validation state.
- clear the error once conditions are satisfied.

So with that in mind, I think the `PlaceOrderButton` should be disabled if the checkout is in error state and submission should only be blocked by that.  The current behaviour in `validateSubmit()` (scrolling to first invalid submit) could be triggered via the processing event callback.

If you're uncertain on how this could be done let me know and I can share more details :)